managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3361,7 +3359,7 @@ private boolean currentLedgerIsFull() {
                 || currentLedgerSize >= (config.getMaxSizePerLedgerMb() * MegaByte));
 
         long timeSinceLedgerCreationMs = clock.millis() - lastLedgerCreatedTimestamp;
-        boolean maxLedgerTimeReached = timeSinceLedgerCreationMs >= maximumRolloverTimeMs;
+        boolean maxLedgerTimeReached = timeSinceLedgerCreationMs >= config.getMaximumRolloverTimeMs();
```

Comment:
I look at the previous logic, there will be a random value here to ensure that all Ledgers are not rollover together at the same time

Reply:
Maybe the random maximum rollover time should be used to control the scheduled task? And use the original configuration to check the ledger whether reach the max rollover time threshold.

Reply:
We have now deleted the scheduled task. If there is no traffic, will it never be cleaned up?

Reply:
https://github.com/apache/pulsar/blob/da66d0e2212c99a5a7e346c38778af0f1276e7d1/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L309

https://github.com/apache/pulsar/blob/da66d0e2212c99a5a7e346c38778af0f1276e7d1/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L3471-L3473

Your current time is fixed, the previous time is floating.
Anyway, I think we need to avoid deleting at the same time. "control the scheduled task" is work for me

Reply:
The `maxLedgerTimeReached` is used to check the ledger should roll over or not, I think it should be the configuration value. And we could use the floating time to control the scheduled task to avoid many rollover operations happen at the same time.

Reply:
The `maxLedgerTimeReached` is used to check the ledger should roll over or not, I think it should be the configuration value. And we could use the floating time to control the scheduled task to avoid many rollover operations happen at the same time.